services/transactionpool/test/add.go

```diff
@@ -46,4 +46,23 @@ var _ = Describe("transaction pool", func() {
 
 	})
 
+	It("dont forward an invalid transaction with gossip", func() {
+
+		tx := test.TransferTransaction().MakeInvalid().Build()
+
+		gossip.When("BroadcastForwardedTransactions", &gossiptopics.ForwardedTransactionsInput{
+			Message: &gossipmessages.ForwardedTransactionsMessage{
+				SignedTransactions: []*protocol.SignedTransaction{tx},
+			},
+		}).Return(&gossiptopics.EmptyOutput{}, nil).Times(0)
+
+		_, err := service.AddNewTransaction(&services.AddNewTransactionInput{
+			SignedTransaction: tx,
+		})
+
+		Expect(err).ToNot(HaveOccurred())
```

Comment:
According to spec we must return error in this case so PublicApi the caller will know the transaction is invalid and tell the user